fastdom.js

```diff
@@ -120,28 +124,51 @@ function scheduleFlush(fastdom) {
 }
 
 /**
- * Runs queued `read` and `write` tasks.
+ * Schedules a new read
+ * batch if one isn't pending.
+ *
+ * @private
+ */
+
+function scheduleReadFlush(fastdom) {
+  if (fastdom.scheduledRead) return;
+  fastdom.scheduledRead = true;
+  if(fastdom._isWriting){
+    setTimeout(fastdom._flushReads, 0);
+  } else {
+    fastdom.immediate.then(fastdom._flushReads);
+    //in case we use setTimeout or a bad Promise polyfill, setTimeout can be called later than a rAF.
+    //by additionally using the ordinary scheduleFlush function we make sure that reads are called as early as possible.
+    scheduleFlush(fastdom);
```

Comment:
> If `rAF` runs first, are we effectively running the same logic as currently on `master`?

Yes, this is one reason why I like this behavior so much. If `read` is called the "new" way (only inside `rAF`) `setTimeout` wins always. If it is called the old way either the enhancement or the exact same old logic wins.

And due to the fact, that I don't need to use `isWriting` flags for using the `setTimeout` approach, `fastdom.read` can also be safely used, if rAF isn't scheduled with `fastdom.write`. So it became more lib agnostic, which was one of the goals.

I think the reason for the race condition isn't heavy work. If there is heavy work to do, both take longer. `setTimeout` and `rAF` are using the same stack/queue, it's just the question where in the "frame life cycle" you call both. This is also the reason, why `setTimeout` called inside a `rAF` always wins, no matter what. (The next frame is 16ms away. But if you call it somewhere else, the next frame might be only 0ms away and then `rAF` always wins.) In normal you will see that one of both always wins in the same situation, but outside `rAF` we can't forecast this.